src/cpu/x64/matmul/brgemm_matmul_utils.hpp

```diff
@@ -270,7 +274,9 @@ struct brgemm_matmul_conf_utils_t {
             // use b_buffer for AMX when:
             // - not bf32 && using non-blocked weights
             // - is bf32
-            return IMPLICATION(!wei_down_convert_to_vnni(), !bgmmc.blocked_B)
+            return IMPLICATION(!(wei_down_convert_to_vnni()
+                                       || (tf32_dt && get_blocked_B())),
+                           !bgmmc.blocked_B)
```

Comment:
Nit: maybe put an implied expression into a dedicated variable? Documentation for this return might require touch up as well.